src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -27,18 +29,12 @@ export const dropdownClassNames = {
   item: `${baseClassName}_item`,
   noSelectedStyles: `${baseClassName}--noSelectedStyles`,
   disabled: `${baseClassName}--disabled`,
+  labelParagraph: `${baseClassName}_label-p`,
 };
 
-export interface DropdownLabelProps extends HtmlLabelProps {}
-
-export interface DropdownItemProps {
-  /** Item value */
-  value: string;
-  /** Item content */
-  children: ReactNode;
-  /** Classname for item */
-  className?: string;
-}
+type BaseListboxPopoverProps = ListboxPopoverProps & {
```

Comment:
Tested these quickly and looks like all Reach related custom types can be removed except OptionalListboxButtonProps. Instead of using [K in key... syntax, Partial<ListboxButtonProps> seems to do the job as well. The refs also seem to be unnecessary and at least styleguidist works without issues if removed.

I would also consider the API at this point, do we really want to support replacing the listbox component or all the related props? Maybe just allow id and className etc. or remove completely?

Reply:
Specifically dropdownButtonProps, dropdownPopoverProps and listboxPopoverComponent. I would consider removing those since we are anyways introducing breaking changes.

Reply:
Left a related comment below as well. Could we just use the styled versions of these and drop the abstraction completely?

Reply:
Cleaned these up quite a bit. A good point about them being mostly obsolete.